### ISSUE TITLE
Setting zoom programatically

### ISSUE THREAD

**Issue by fredludlow** [arose/ngl#240]
Another quick question:
I can use `Stage.setOrientation` to restore a particular camera position/view (from Stage.`getOrientation`) but is there any way to query and set the zoom?

**Comment by arose** [comment]
There are undocumented (and subject to change) set methods in the Viewer class.
- `stage.viewer.rotate(axis, angle)` // axis is a Vector3
- `stage.viewer.zoom(distance, set)`  // if set is false the distance will be added to the current one
- `stage.viewer.center(position)`  // position is a Vector3

There are no query/get methods.

The whole camera setup needs to be overhauled...

**Comment by fredludlow** [comment]
Woop, it works! Thanks!
In case anyone else is interested:
```
getZoomDistance: function() { 
    return stage.viewer.camera.position.clone()
           .sub( stage.viewer.controls.target )
           .length();
}
```
Then `stage.viewer.zoom( distance, true )` to restore

**Comment by fredludlow** [comment]
I'm trying to update my code that saves and restores the orientation. I'm now using:

    var orientation = stage.viewerControls.getOrientation(); // Contains 4x4 matrx as described in viewer-controls.js
    stage.viewerControls.orient( orientation ); // Apply 

For orthographic camera I also need to insert a call to 

    stage.viewer.updateZoom(); 
    stage.viewer.requestRender();

I think the `orient` method is missing a call to `this.viewer.updateZoom()` at line [126](https://github.com/arose/ngl/blob/a8f838928f7af1c9409f6e64f8a904584a17c2d4/src/controls/viewer-controls.js#L126) of viewer-controls.js? 

Or...  am I using the wrong API for this and if so how should I do this?
(Basically want to recreate the exact camera/view position from a previous point)

**Comment by arose** [comment]
It is the correct API, it was broken, hopefully fixed with b232333120bad0f842a0fdfb740625541b901ea4 and 0fc4096247166d26f99d05d05071e2be3d7f3002

**Comment by fredludlow** [comment]
Yup, that fixes it, thanks!
